pkg/generators/models/testdata/cases/allof_enum/api.yaml

```diff
@@ -9,14 +9,14 @@ components:
       type: object
       properties:
         leftHand:
```

Comment:
is the old behavior also still working? I.e. an allof of one ref and one nullable:true?

Reply:
Give a concrete example

Reply:
This file, but without your commits, basically the master version of this testcase.

Reply:
It works , in that it will generate, but once you have multiple values the all of it will flatten the to a "primitive" type instead of using references to named types. In this case you would get pointers to strings instead of pointers to the enums

Reply:
@trusch  To expand on my comment and make it more concrete. This specific test case was created to generate a pointer to Anything. The expected code is this:

```go
type Artist struct {
    leftHand *Anything
    rightHand *Color
}
```

To correctly handle the `allOf` with multiple references, we now produce a flat inline `struct` or primitive value instead of a named struct _but only if_ there is more than one item in the `allOf`.  See this line https://github.com/contiamo/openapi-generator-go/pull/48/files/80e9188dd7cd753f3a8db5a39932f539d6126e69#diff-4413a110e8bea2cc5eb98517239e35b900fa50874538f2a16f14cff348e7540dR241

This means that 

```yaml
  allOf:
    - $ref: "#/components/schemas/AnyThing"
    - nullable: true
```
will produce `*string` instead of `*Anything`.  This is because it actually merges the definition of Anything with the `nullable:true`, but it strips out the reference to Anything. So the resulting type becomes "anonymous" or inlined.

But if you update the schema so that the nullable is defined at the same level as the `allOf`, then it will become the special case handled here https://github.com/contiamo/openapi-generator-go/pull/48/files/80e9188dd7cd753f3a8db5a39932f539d6126e69#diff-4413a110e8bea2cc5eb98517239e35b900fa50874538f2a16f14cff348e7540dR241

meaning, this schema

```yaml
nullable: true
allOf:
  - $ref: "#/components/schemas/AnyThing"
```

This is why I changed the schema here. The expected/output Go code does not change, but only if we make this  schema change. 
actually produces the desired `*Anything`

Reply:
@trusch you asked about how this looks when we run it on hub. I noticed a breaking change for Hub for the life cycle status

```diff
diff --git pkg/openapi/models/model_file_storage_update_request.go pkg/openapi/models/model_file_storage_update_request.go
index 8d82d861..f861d968 100644
--- pkg/openapi/models/model_file_storage_update_request.go
+++ pkg/openapi/models/model_file_storage_update_request.go
@@ -23,7 +23,7 @@ type FileStorageUpdateRequest struct {
        Icon *string `json:"icon,omitempty"`
        // LifecycleStatus: The lifecycle state of a resource.
        // This will default to 'active' when not specified on resource creation.
-       LifecycleStatus *LifecycleStatus `json:"lifecycleStatus,omitempty"`
+       LifecycleStatus *string `json:"lifecycleStatus,omitempty"`
        // Properties:
        Properties map[string]interface{} `json:"properties,omitempty"`
 }
@@ -37,9 +37,6 @@ func (m FileStorageUpdateRequest) Validate() error {
                "displayName": validation.Validate(
                        m.DisplayName, validation.Length(0, 63),
                ),
-               "lifecycleStatus": validation.Validate(
-                       m.LifecycleStatus,
-               ),
                "properties": validation.Validate(
                        m.Properties,
                ),
@@ -97,12 +94,12 @@ func (m *FileStorageUpdateRequest) SetIcon(val *string) {
 }
```


```diff
-
-// DB2DataSourceConnectionInfo is an object. Contains parameters for connecting to the data source.
-// If a parameter is not required and not set it will be replaced by the documented default value if any.
+// DB2DataSourceConnectionInfo is an object. DB2 connection properties
 type DB2DataSourceConnectionInfo struct {
        // Database:
        Database string `json:"database"`
diff --git pkg/openapi/models/model_db2_product.go pkg/openapi/models/model_db2_product.go
index 9969998c..45d5b0c3 100644
--- pkg/openapi/models/model_db2_product.go
+++ pkg/openapi/models/model_db2_product.go
@@ -15,7 +15,7 @@ type DB2Product struct {
        ProductName DB2ProductProductName `json:"productName"`
        // Properties: Contains parameters for connecting to the data source.
        // If a parameter is not required and not set it will be replaced by the documented default value if any.
-       Properties BaseConnectionInfo `json:"properties"`
+       Properties map[string]interface{} `json:"properties"`
 }
 
 // Validate implements basic validation for this model
@@ -41,11 +41,11 @@ func (m *DB2Product) SetProductName(val DB2ProductProductName) {
 }
 
 // GetProperties returns the Properties property
-func (m DB2Product) GetProperties() BaseConnectionInfo {
+func (m DB2Product) GetProperties() map[string]interface{} {
        return m.Properties
 }
 
 // SetProperties sets the Properties property
-func (m *DB2Product) SetProperties(val BaseConnectionInfo) {
+func (m *DB2Product) SetProperties(val map[string]interface{}) {
        m.Properties = val
 }
```

Reply:
the second diff is actually fixing a bug in the type generation

Reply:
I guess we can fix the first issue by restructuring how we inject the nullable in the api spec

Reply:
yeah i need to double check that. The second change is actually the fix, the original code was _broken_

Reply:
Yes, each of the lifecycle status definitions look like 
```yaml
    lifecycleStatus:
          allOf:
            - $ref: "#/components/schemas/LifecycleStatus"
            - nullable: true
```

so we can update to 

```yaml
    lifecycleStatus:
          nullable: true
          allOf:
            - $ref: "#/components/schemas/LifecycleStatus"
```

Reply:
Here are the 12 spots we need to update 
![image](https://user-images.githubusercontent.com/891889/127640290-38d98b19-e25a-4f47-a71e-b46047de3dbf.png)